package/kedro_viz/api/rest/responses.py

```diff
@@ -273,19 +330,154 @@ def get_default_response() -> GraphAPIResponse:
     )
 
 
-class EnhancedORJSONResponse(ORJSONResponse):
-    @staticmethod
-    def encode_to_human_readable(content: Any) -> bytes:
-        """A method to encode the given content to JSON, with the
-        proper formatting to write a human-readable file.
+def get_node_metadata_response(node_id: str):
+    """API response for `/api/nodes/node_id`."""
+    node = data_access_manager.nodes.get_node_by_id(node_id)
+    if not node:
+        return JSONResponse(status_code=404, content={"message": "Invalid node ID"})
 
-        Returns:
-            A bytes object containing the JSON to write.
+    if not node.has_metadata():
+        return JSONResponse(content={})
 
-        """
-        return orjson.dumps(
-            content,
-            option=orjson.OPT_INDENT_2
-            | orjson.OPT_NON_STR_KEYS
-            | orjson.OPT_SERIALIZE_NUMPY,
+    if isinstance(node, TaskNode):
+        return TaskNodeMetadata(node)
+
+    if isinstance(node, DataNode):
+        return DataNodeMetadata(node)
+
+    if isinstance(node, TranscodedDataNode):
+        return TranscodedDataNodeMetadata(node)
+
+    return ParametersNodeMetadata(node)
+
+
+def get_selected_pipeline_response(registered_pipeline_id: str):
+    """API response for `/api/pipeline/pipeline_id`."""
+    if not data_access_manager.registered_pipelines.has_pipeline(
+        registered_pipeline_id
+    ):
+        return JSONResponse(status_code=404, content={"message": "Invalid pipeline ID"})
+
+    modular_pipelines_tree = (
+        data_access_manager.create_modular_pipelines_tree_for_registered_pipeline(
+            registered_pipeline_id
+        )
+    )
+
+    return GraphAPIResponse(
+        nodes=data_access_manager.get_nodes_for_registered_pipeline(  # type: ignore
+            registered_pipeline_id
+        ),
+        edges=data_access_manager.get_edges_for_registered_pipeline(  # type: ignore
+            registered_pipeline_id
+        ),
+        tags=data_access_manager.tags.as_list(),
+        layers=data_access_manager.get_sorted_layers_for_registered_pipeline(
+            registered_pipeline_id
+        ),
+        pipelines=data_access_manager.registered_pipelines.as_list(),
+        selected_pipeline=registered_pipeline_id,
+        modular_pipelines=modular_pipelines_tree,  # type: ignore
+    )
+
+
+def get_package_version(package_name: str):
```

Comment:
I'm not quite sure if all the methods her belong in this file. As stated at the top this `responses.py` is meant for: `kedro_viz.api.rest.responses defines REST response types.` Again, I haven't looked at the architecture for a while, but this definitely seems like it's not supposed to be here.